sdk/core/core-https/src/util/userAgent.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { setPlatformSpecificData, getHeaderName } from "./userAgentPlatform";
+import { SDK_VERSION } from "../constants";
+
+function getUserAgentString(telemetryInfo: Map<string, string>): string {
+  const parts: string[] = [];
+  for (const [key, value] of telemetryInfo) {
+    const token = value ? `${key}/${value}` : key;
+    parts.push(token);
+  }
+  return parts.join(" ");
+}
+
+export function getUserAgentHeaderName(): string {
+  return getHeaderName();
+}
+
+export function getUserAgentValue(prefix?: string): string {
```

Comment:
General nit: I don't know if we've updated the version of TypeDoc we use and if this is no longer a problem, but we've seen that anything that's exported, even if it isn't exported from the entry point, still shows up in docs unless we add the `@ignores` and `@internal` tags.

Reply:
Sprinkled these liberally throughout the internal utilities. Will have to check the tsdoc output later.